idaes/apps/grid_integration/multiperiod/design_and_operation_models.py

```diff
@@ -0,0 +1,122 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+
+from pyomo.environ import (
+    Var,
+    Param,
+    Binary,
+    Expression,
+    NonNegativeReals,
+    Constraint,
+)
+from functools import reduce
+from idaes.core.base.process_base import declare_process_block_class
+from idaes.models.unit_models import SkeletonUnitModelData
+from pyomo.common.config import ConfigValue, In
+
+
+@declare_process_block_class("DesignModel")
+class DesignModelData(SkeletonUnitModelData):
```

Comment:
Out of interest, why are you inheriting `SkeletonUnitModelData` here and not `UnitModelData` (or another base class)?

Reply:
I believe this was the first thing that came to mind when the original development was happening. @radhakrishnatg can probably comment more on this.

Reply:
@andrewlee94 From what I remember, `UnitModelData` class has several additional methods that I think are not needed. I just need a simple Pyomo block. If `UnitModelData` is a better alternative, then @djlaky, please make the change.

Reply:
It might be OK, I was mostly curious (and thought that SkeletonUnitModel might have inherited from UnitModelBlock anyway, but I just checked and it does not). However, do you use the methods defined on SkeletonUnitModel at all (add_ports, fix_initialization_states, initialize (deprecated)), or would ProcessBlock be just as good (the parent of SkeletonUnitModel)?